hoot-rnd/src/main/cpp/hoot/rnd/conflate/network/IndexedEdgeMatchSet.cpp

```diff
@@ -83,30 +155,214 @@ EdgeMatchPtr IndexedEdgeMatchSet::getMatch(const EdgeMatchPtr &em) const
+{
+  QSet<ConstEdgeMatchPtr> result;
+  QList<ConstEdgeMatchPtr> list;
+  QSet<ConstEdgeMatchPtr> candidates;
 
-  foreach (const EdgeMatchPtr& em, _matches.keys())
+  foreach (const EdgeString::EdgeEntry& ee, e->getString1()->getAllEdges())
   {
-    if (em->getString1()->getFrom() == v1 &&
-      em->getString2()->getFrom() == v2)
+    foreach(const ConstEdgeMatchPtr em, _edgeToMatch[ee.getEdge()])
+    {
+      list.append(em);
+    }
+  }
+  foreach (const EdgeString::EdgeEntry& ee, e->getString2()->getAllEdges())
+  {
+    foreach(const ConstEdgeMatchPtr em, _edgeToMatch[ee.getEdge()])
+    {
+      list.append(em);
+    }
+  }
+
+  candidates = list.toSet();
+
+  foreach (const ConstEdgeMatchPtr& em, candidates)
+  {
+    if (em->getString1()->overlaps(e->getString1()) ||
+      em->getString2()->overlaps(e->getString2()))
+    {
+      result.insert(em);
+    }
+  }
+
+  return result;
+}
+
+QSet<ConstEdgeMatchPtr> IndexedEdgeMatchSet::getMatchesWithInteriorVertex(ConstNetworkVertexPtr v)
+  const
+{
+  QSet<ConstEdgeMatchPtr> result;
+  foreach (const ConstEdgeMatchPtr& em,
+//    _matches.keys())
+    _vertexToMatch[v])
+  {
+    if (em->getString1()->containsInteriorVertex(v) ||
+      em->getString2()->containsInteriorVertex(v))
+    {
+      result.insert(em);
+    }
+  }
+
+  return result;
+}
+
+QSet<ConstEdgeMatchPtr> IndexedEdgeMatchSet::getMatchesWithTermination(ConstNetworkVertexPtr v1,
+  ConstNetworkVertexPtr v2) const
+{
+  QSet<ConstEdgeMatchPtr> result;
+
+  // go through all the matches that contain v1 and v2
+  foreach (const ConstEdgeMatchPtr& em,
+//    _matches.keys())
+    _vertexToMatch[v1] & _vertexToMatch[v2])
+  {
+    if (em->getString1()->isFromOnVertex() &&
+      em->getString1()->getFromVertex() == v1 &&
+      em->getString2()->isFromOnVertex() &&
+      em->getString2()->getFromVertex() == v2)
     {
       result.insert(em);
     }
-    else if (em->getString1()->getTo() == v1 &&
-      em->getString2()->getTo() == v2)
+    else if (em->getString1()->isToOnVertex() &&
+      em->getString1()->getToVertex() == v1 &&
+      em->getString2()->isToOnVertex() &&
+      em->getString2()->getToVertex() == v2)
     {
       result.insert(em);
     }
+    else if (em->getString1()->isFromOnVertex() &&
+      em->getString1()->getFromVertex() == v2 &&
+      em->getString2()->isFromOnVertex() &&
+      em->getString2()->getFromVertex() == v1)
+    {
+      result.insert(em);
+    }
+    else if (em->getString1()->isToOnVertex() &&
+      em->getString1()->getToVertex() == v2 &&
+      em->getString2()->isToOnVertex() &&
+      em->getString2()->getToVertex() == v1)
+    {
+      result.insert(em);
+    }
+  }
+
+  return result;
+}
+
+QSet<ConstEdgeMatchPtr> IndexedEdgeMatchSet::getConnectingStubs(ConstEdgeMatchPtr a,
+  ConstEdgeMatchPtr b) const
+{
+  QSet<ConstEdgeMatchPtr> result;
+
+  if (a->containsStub() || b->containsStub())
+  {
+    // no connecting stubs
```

Comment:
No, that was old functionality that was never removed. I'm updating the code to and comments to clarify.